### PR TITLE
[Datafactory] Add Webhook Activity to Azure Data Factory 

### PR DESCRIPTION
**Pull request by arsunda**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 1]
```diff
@@ -2540,6 +2540,97 @@
...
+      "x-ms-enum": {
+        "name": "WebHookActivityMethod",
+        "modelAsString": true
+      }
+    },
+    "WebHookActivityAuthentication": {
+      "description": "WebHook activity authentication properties.",
+      "properties": {
+        "type": {
+          "description": "WebHook activity authentication (Basic/ClientCertificate/MSI)",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: You should make this one enum too

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 2]
```diff
@@ -2540,6 +2540,97 @@
...
+        "username": {
+          "description": "WebHook activity authentication user name for basic authentication.",
+          "type": "string"
+        },
+        "password": {
+          "description": "Password for the PFX file or basic authentication.",
+          "$ref": "../datafactory.json#/definitions/SecureString"
+        },
+        "resource": {
+          "description": "Resource for which Azure Auth token will be requested when using MSI Authentication.",
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Is the value here ARM resource ID?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json (hunk 1 above)
Comment (arsunda) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: This is the same as WebActivityAuthentication, copied over and renamed. I was going to reference that definition directly but didnt do so incase there would be some differences in the future. Should i still change it to enum?

Comment (arsunda) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Changed this as per comments

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json [hunk 3]
```diff
@@ -2469,12 +2469,25 @@
...
+        "modelAsString": true
+      }
+    },
     "WebActivityAuthentication": {
       "description": "Web activity authentication properties.",
       "properties": {
         "type": {
           "description": "Web activity authentication (Basic/ClientCertificate/MSI)",
-          "type": "string"
+          "$ref": "#/definitions/WebActivityAuthenticationType"
```
Comment (hvermis) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: @jhendrixMSFT, Is this considered a breaking change? The new type is an enum that is modeled as a string so in SDK it should be still string.

**Review by hvermis — APPROVED** [review]

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json (hunk 3 above)
Comment (arsunda) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Changing it back to old format

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**